src/state.py

```diff
@@ -85,21 +93,63 @@ def validate_hostname(cls, value: str) -> str:
     def validate_path(cls, value: str) -> str:
         """Validate the path.
 
+        Args:
+            value: The value to validate.
+
+        Returns:
+            The value after validation.
+        """
+        return validate_path_value(value)
+
+    @pydantic.field_validator("fail_timeout")
+    @classmethod
+    def validate_fail_timeout(cls, value: str) -> str:
+        """Validate the fail_timeout.
+
+        Args:
+            value: The value to validate.
+
+        Returns:
+            The value after validation.
+        """
+        check_nginx_time_str(value)
+        return value
+
+    @pydantic.field_validator("backends_path")
+    @classmethod
+    def validate_backends_path(cls, value: str) -> str:
+        """Validate the backends_path.
+
+        Args:
+            value: The value to validate.
+
+        Returns:
+            The value after validation.
+        """
+        return validate_path_value(value)
+
+    @pydantic.field_validator("proxy_cache_valid")
+    @classmethod
+    def validate_proxy_cache_valid(cls, value: str) -> str:
+        """Validate the proxy_cache_valid.
+
         Args:
             value: The value to validate.
 
         Raises:
-            ValueError: The validation failed.
+            ValueError: The proxy_cache_valid is invalid.
 
         Returns:
             The value after validation.
         """
-        # This are the valid characters for path in addition to `/`:
-        # a-z A-Z 0-9 . - _ ~ ! $ & ' ( ) * + , ; = : @
-        # https://datatracker.ietf.org/doc/html/rfc3986#section-3.3
-        valid_path = re.compile(r"[/\w.\-~!$&'()*+,;=:@]+", re.IGNORECASE)
-        if valid_path.fullmatch(value) is None:
-            raise ValueError("Path contains non-allowed character")
+        for item in value:
+            tokens = item.split(" ")
```

Comment:
Nit: Would whitespace separator not suffice here? (Would you be able to add the valid string example format just in case perhaps later as a follow up)

Reply:
The format is defined by nginx. For single space as separator, I have tested and confirmed it works.
I am not sure if other whitespace or multiple whitespace would work with the nginx configuration.

I can add this a backlog item, but I think this is not high-priority right now.